### PR TITLE
feat(version): expand v1 update policy to cover all versions

### PR DESCRIPTION
**Pull request by acburdine**
no issue
- previously, the supported version policy was you could update to any new major if you were on the latest v1
- now, we're expanding that to say you can upgrade to any new major only if you're on the latest version of the current major

### CODE REVIEW

File: lib/utils/version.js [hunk 1]
```diff
@@ -115,11 +115,14 @@ const utils = {
...
+        const latestMajor = latestMajorVersions[`v${activeMajor}`];
+
+        if (activeVersion !== latestMajor && semver.diff(activeVersion, versionToUse) === 'major') {
             const majorToUse = semver.major(versionToUse);
+
             throw new CliError({
-                message: `You are trying to update to Ghost v${majorToUse}, but your blog is not on the latest Ghost 1.0 version`,
-                help: 'Instead run "ghost update v1".'
+                message: `You are trying to update to Ghost v${majorToUse}, but your blog is not on the latest Ghost ${activeMajor}.0 version`,
+                help: `First upgrade to the latest Ghost v${activeMajor} with "ghost update v${activeMajor}", then update to Ghost ${majorToUse} with "ghost update".`
```
Comment (daniellockyer) [inline lib/utils/version.js:125]: I would nuke the first word personally:

```suggestion
                help: `Upgrade to the latest Ghost v${activeMajor} with "ghost update v${activeMajor}", then update to Ghost ${majorToUse} with "ghost update".`
```

Comment (acburdine) [reply lib/utils/version.js]: done

File: lib/utils/version.js [hunk 2]
```diff
@@ -115,11 +115,14 @@ const utils = {
...
-        if (activeMajor === 1 && activeVersion !== latestV1 && semver.diff(activeVersion, versionToUse) === 'major') {
+        const latestMajor = latestMajorVersions[`v${activeMajor}`];
+
+        if (activeVersion !== latestMajor && semver.diff(activeVersion, versionToUse) === 'major') {
             const majorToUse = semver.major(versionToUse);
+
             throw new CliError({
-                message: `You are trying to update to Ghost v${majorToUse}, but your blog is not on the latest Ghost 1.0 version`,
-                help: 'Instead run "ghost update v1".'
+                message: `You are trying to update to Ghost v${majorToUse}, but your blog is not on the latest Ghost ${activeMajor}.0 version`,
```
Comment (daniellockyer) [inline lib/utils/version.js:124]: Would this be a more correct message?

```suggestion
                message: `You are trying to update to Ghost v${majorToUse}, but your blog is not on the latest Ghost v${activeMajor} version`,
```

Comment (acburdine) [reply lib/utils/version.js:124]: yup, would definitely be more consistent - I'll fix

Comment (acburdine) [reply lib/utils/version.js:124]: or maybe the second part might be better worded "but your blog is not running the latest Ghost v{activeMajor}"

**Comment by ErisDS** [comment]
How's this?

```
message: `Unable to update Ghost directly to v{versionToUse}. You must be on the latest v${activeMajor}.x to update across major versions.`,
help: `Run "ghost update v${activeMajor}" to get the latest v${activeMajor}.x version, then run "ghost update" to get to v{majorToUse}.`
```

Full example: 

Message: Unable to update Ghost directly to v4.7.1. You must be on the latest v2.x to update across major versions.
Help: Run "ghost update v2" to get the latest v2.x version, then run "ghost update" to get to Ghost v4

**Comment by acburdine** [comment]
@ErisDS yup, that wording's more clear - will use that 😄

**Pull request merged**